autoload/lsc/reference.vim

```diff
@@ -143,15 +145,65 @@ function! s:showHover(result) abort
       let l:lines += split(item, "\n")
     endif
   endfor
-  if get(g:, 'lsc_hover_popup', v:true) && exists('*popup_atcursor')
-    call popup_clear()
-    call popup_atcursor(l:lines, {
+  if get(g:, 'lsc_hover_popup', v:true) 
+        \ && (exists('*popup_atcursor') || exists('*nvim_open_win'))
+    call s:closeHoverPopup()
+    call s:openHoverPopup(l:lines)
+  else
+    call lsc#util#displayAsPreview(lines, function('lsc#util#noop'))
+  endif
+endfunction
+
+function! s:openHoverPopup(lines)
+  " Sanity check, if there is no hover text then don't waste resources creating an
+  " empty popup.
+  if len(a:lines) == 0
+    return
+  endif
+  if has("nvim")
+    let buf = nvim_create_buf(v:false, v:true)
+    " Note, the +2s below will be used for padding around the hover text.
+    let height = len(a:lines) + 2
+    let width = 1
+    " Need to figure out the longest line and base the popup width on that.
+    for val in a:lines
+      if len(val) + 2 > l:width
```

Comment:
Should this be `strdisplaywidth`?

Reply:
I plead ignorant of strdisplaywidth.

My quick Googling did not yield much.

What does it do?
I do note my padding is slightly ugly and hacky. If it can improved, I welcome it.

Do you believe strdisplaywidth can help?

Reply:
It will give the number of characters that are used on the screen, which may be more (for full width characters) or less (for multibyte characters) than the number of bytes in the string which is what `len()` gives.

For example:

- A tab character: `len("\t")` is `1` while `strdisplaywidth("\t")` may be `4` or `8` depending on settings and which column it starts on.
- A full width character: `len("ア")` is `3` while `strdisplaywidth("ア")` is `2`.

If I have the text:

```
1234abc
アイabc
```

These will have different `len`, but the same `strdisplaywidth` and the `abc` from each will be lined up (github doesn't show it this way, but vim/nvim should). If we're trying to figure out how much room we need to show a line, `strdisplaywidth` is what we want.

Reply:
Yes, I understand now.

Indeed, you are correct.

I will update with that enhancement soon.